BasicTypesProxy.cxx

```diff
@@ -24,6 +24,91 @@ namespace
       return x == y;
     }
   }
+
+  /// Helper class to track inf/nans encountered.
+  class InfNanTable
+  {
+    private:
+      struct Encounters
+      {
+        std::size_t count = 0;  /// total number of times this var saw a NaN/inf
+        std::string firstFile;  /// first file a NaN/inf was seen in
+        std::size_t firstEntry = std::numeric_limits<std::size_t>::max(); /// entry number within firstFile where the first NaN/inf was seen
+      };
+
+    public:
+      ~InfNanTable() { EmitTable(); }
+
+      void LogInf(const std::string& varPath, const std::string& file, std::size_t entry)  { Log(fInfEncounters, varPath, file, entry); CheckAbort(); };
+      void LogNaN(const std::string& varPath, const std::string& file, std::size_t entry)  { Log(fNaNEncounters, varPath, file, entry); CheckAbort(); };
+
+      void EmitTable(std::ostream & stream = std::cerr) const
+      {
+        bool showedHeader = false;
+        bool anyWarns = false;
```

Comment:
I'm not sure this actually adds anything at the moment, anyways---I'll leave this to some future person who decides they want to write this out to a file instead of `cerr`.  (In that case you'd want to actually _do_ something with the fact that it's throwing an exception, rather than just ignoring it.)  The consequences of `ostream` throwing in this case is just that the program will crash and end slightly sooner, but if `cerr` is throwing an exception that's probably because its buffer is full, and we've got bigger problems in that case.